src/app/_components/profile/UserDetails/UserDetailsDisplay.tsx

```diff
@@ -5,49 +5,87 @@ import {
   CardFooter,
   CardHeader,
   CardTitle,
-} from "../../shadcn/card";
-import { cn } from "~/lib/utils";
+} from "@components/card";
 import { api } from "~/trpc/react";
 import { useSession } from "next-auth/react";
 import DetailsEditDialog from "./DetailsEditDialog";
-import ProfilePictureDisplay from "~/app/_components/home/ProfilePicture/ProfilePictureDisplay";
+import ImageGrid from "~/app/_components/signup/ImageGrid";
+import React from "react";
+import { Skeleton } from "@components/skeleton";
 
 const UserDetailsDisplay: React.FC = () => {
   const session = useSession();
 
-  const { data: userDetails } = api.user.findUserById.useQuery(
-    session.data?.user?.id ?? "",
-    { enabled: !!session.data },
-  );
+  const { data: userDetails, isLoading: isLoadingUserDetails } =
+    api.user.findUserById.useQuery(session.data?.user?.id ?? "", {
+      enabled: !!session.data,
+    });
 
   const facts = [
     `I live in ${userDetails?.city}`,
     `I am a ${userDetails?.gender}.`,
     `I am looking for a ${userDetails?.sexualPreference}!`,
   ];
+  const {
+    data: usersImages,
+    refetch: refetchImages,
+    isLoading: isLoadingImages,
+  } = api.image.findMany.useQuery(session.data?.user?.id ?? "", {
+    enabled: !!session.data,
+  });
 
   return (
     <Card className="max-w-screen-sm">
       <CardHeader>
-        <CardTitle>{`${userDetails?.name}, ${userDetails?.age}`}</CardTitle>
-        <CardDescription>{userDetails?.email}</CardDescription>
+        {isLoadingUserDetails || !userDetails ? (
+          <div className="space-y-2">
+            <Skeleton className="h-8 w-1/5 rounded-md" />
+            <Skeleton className="h-4 w-1/6 rounded-md" />
+          </div>
+        ) : (
+          <>
+            <CardTitle>{`${userDetails?.name}, ${userDetails?.age}`}</CardTitle>
+            <CardDescription>{userDetails?.email}</CardDescription>
+          </>
+        )}
```

Comment:
consider getting the condition out from the react section and put it in variable and than put the variabe inside